src/write_gcm.c

```diff
@@ -3105,6 +3146,64 @@ static void wg_json_Metric(json_ctx_t *jc,
   wg_json_map_close(jc);
 }
 
+static int wg_default_metric_value_type_from_ds_type(int ds_type) {
+  int metric_value_type = METRIC_VALUE_TYPE_UNKNOWN;
+
+  switch(ds_type) {
+    case DS_TYPE_GAUGE:
+      metric_value_type = METRIC_VALUE_TYPE_DOUBLE;
+      break;
+    case DS_TYPE_DERIVE:
+    case DS_TYPE_COUNTER:
+      metric_value_type = METRIC_VALUE_TYPE_INT64;
+      break;
+  }
+
+  return metric_value_type;
+}
+
+static int wg_metric_value_type_from_wg_payload_t(
+    const wg_payload_t *element) {
+  int metric_value_type = METRIC_VALUE_TYPE_UNKNOWN;
+
+  for (int i = 0; i < element->key.num_metadata_entries; ++i) {
+    wg_metadata_entry_t *entry = &element->key.metadata_entries[i];
+    if (strcmp(entry->key, custom_metric_value_type_key) == 0) {
+      if (strcmp(entry->value.value_text, "INT64") == 0) {
```

Comment:
This could be a loop over the `metric_value_type_strings` array, right?

Reply:
Not done.

Reply:
I'm torn on this, I feel as though it might be too magical by using `metric_value_type_strings`, here's the code I would use in its place, and I'm not sure that the meaning is clear to the end user.

```
      for (int j = 0; j < STATIC_ARRAY_SIZE(metric_value_type_strings); j++) {
        if (strcmp(entry->value.value_text, metric_value_type_strings[j]) == 0) {
          return j;
        }
      }
      
      WARNING("write_gcm: ignoring metric value type provided: %s, will use"
              "default instead.", entry->value.value_text);
      break;
```

WDYT? Is there a better more intuitive way to use the array without losing meaning? If you feel that this is clear then I'll push a commit for it.

Reply:
@igorpeshansky friendly ping for your thoughts here. Looking to validate whether the proposed solution is still "developer friendly", or if we should stay with the current implementation due to its explicitness.